### DISCUSSION TITLE
HAR files for @xstate/inspect

### DISCUSSION THREAD

**Discussion by RafalFilipek** [statelyai/xstate#1522] [Ideas]
Hi, Imagine you can export current inspector state with events, states, sequences as something like `har` file. Importing such file could trigger read only mode for inspector.  

In my project I have the huge number of cases with context (user context) data.

What you think about `xhar` files idea?

Thank you

**Comment by davidkpiano** [comment]
That's an interesting idea! I've been telling people (more often now) that, for restoring states, **event sourcing** should be used over a state snapshot. This has lots of advantages, including the natural ability to restore invoked/spawned actors as well (since that's just how events work - you replay the events exactly, you should get the same outcome).
